### PR TITLE
Add arm64 support

### PR DESCRIPTION
**Pull request by krmbzds**
This adds support for 64-bit ARM devices. Tested on my Raspberry Pi 4 Model B. :slightly_smiling_face:

### CODE REVIEW

**Review by SuperQ — APPROVED** [review]

File: bin/download [hunk 1]
```diff
@@ -13,20 +13,30 @@ fail() {
...
+  Darwin) echo "macOS" ;;
+  *) echo "${uname}" ;;
   esac
 }
 
-filename="vale_${ASDF_INSTALL_VERSION}_$(get_platform)_64-bit.tar.gz"
+get_arch() {
+  local -r machine="$(uname -m)"
+
+  if [[ ${machine} == "arm64" ]] || [[ ${machine} == "aarch64" ]]; then
```
Comment (SuperQ) [inline bin/download:24]:
```suggestion
  if [[ "${machine}" == "arm64" ]] || [[ "${machine}" == "aarch64" ]]; then
```

File: bin/download [hunk 2]
```diff
@@ -13,20 +13,30 @@ fail() {
 get_platform() {
   local uname="$(uname)"
   case "${uname}" in
-    Darwin) echo "macOS" ;;
-    *) echo "${uname}" ;;
+  Darwin) echo "macOS" ;;
+  *) echo "${uname}" ;;
```
Comment (SuperQ) [inline bin/download:17]: https://google.github.io/styleguide/shellguide.html#case-statement

```suggestion
    Darwin) echo "macOS" ;;
    *) echo "${uname}" ;;
```

**Review by SuperQ — CHANGES_REQUESTED** [review]
Minor shell nits.

File: bin/download [hunk 3]
```diff
@@ -18,15 +18,25 @@ get_platform() {
...
 }
 
-filename="vale_${ASDF_INSTALL_VERSION}_$(get_platform)_64-bit.tar.gz"
+get_arch() {
+  local -r machine="$(uname -m)"
+
+  if [[ "${machine}" == "arm64" ]] || [[ "${machine}" == "aarch64" ]]; then
+    echo "arm64"
+  else
+    echo "64-bit"
```
Comment (osg) [inline bin/download:27]: Is this always going to return 64-bit? Never 32-bit?

Comment (krmbzds) [reply bin/download:27]: Yes, it will always return 64-bit unless its an arm64 device. Previously `64-bit` value was hardcoded in the URL.

Comment (krmbzds) [reply bin/download:27]: Vale has no `32-bit` release and `asdf-vm` does not support Windows. So this should cover all supported release types for Vale:

- [vale_2.20.2_Linux_64-bit.tar.gz](https://github.com/errata-ai/vale/releases/download/v2.20.2/vale_2.20.2_Linux_64-bit.tar.gz)
- [vale_2.20.2_Linux_arm64.tar.gz](https://github.com/errata-ai/vale/releases/download/v2.20.2/vale_2.20.2_Linux_arm64.tar.gz)
- [vale_2.20.2_macOS_64-bit.tar.gz](https://github.com/errata-ai/vale/releases/download/v2.20.2/vale_2.20.2_macOS_64-bit.tar.gz)
- [vale_2.20.2_macOS_arm64.tar.gz](https://github.com/errata-ai/vale/releases/download/v2.20.2/vale_2.20.2_macOS_arm64.tar.gz)
- <del>[vale_2.20.2_Windows_64-bit.zip](https://github.com/errata-ai/vale/releases/download/v2.20.2/vale_2.20.2_Windows_64-bit.zip)</del>

**Review by SuperQ — APPROVED** [review]

**Comment by SuperQ** [comment]
Thanks!

**Review by osg — APPROVED** [review]

**Pull request merged**